backend/auth/handlers.go

```diff
@@ -0,0 +1,340 @@
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorUpdateUserAccount})
+		return
+	}
+	dbUser.Password = string(hashedPassword)
+	filter := bson.M{"username": dbUser.Username}
+	_, err = dbadapter.WebuiDBClient.RestfulAPIPost(UserAccountDataColl, filter, configmodels.ToBsonM(dbUser))
+	if err != nil {
+		logger.DbLog.Errorln(err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorUpdateUserAccount})
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{})
+}
+
+func Login(jwtSecret []byte) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var inputUser configmodels.User
+		err := c.ShouldBindJSON(&inputUser)
+		if err != nil {
+			logger.AuthLog.Errorln(err.Error())
+			c.JSON(http.StatusBadRequest, gin.H{"error": errorInvalidDataProvided})
+			return
+		}
+		if inputUser.Username == "" {
+			c.JSON(http.StatusBadRequest, gin.H{"error": errorMissingUsername})
+			return
+		}
+		if inputUser.Password == "" {
+			c.JSON(http.StatusBadRequest, gin.H{"error": errorMissingPassword})
+			return
+		}
+		dbUser, err := fetchUser(inputUser.Username)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": errorRetrieveUserAccount})
+			return
+		}
+		if dbUser == nil {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": errorIncorrectCredentials})
+			return
+		}
+		if err = bcrypt.CompareHashAndPassword([]byte(dbUser.Password), []byte(inputUser.Password)); err != nil {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": errorIncorrectCredentials})
+			return
+		}
+		jwt, err := generateJWT(dbUser.Username, dbUser.Permissions, jwtSecret)
+		if err != nil {
+			logger.AuthLog.Errorln(err.Error())
+			c.JSON(http.StatusInternalServerError, gin.H{"error": errorLogin})
+			return
+		}
+		c.JSON(http.StatusOK, gin.H{"token": jwt})
+	}
+}
+
+func IsFirstAccountIssued() (bool, error) {
+	users, err := fetchUsers()
+	if err != nil {
+		return false, err
+	}
+	return len(users) > 0, nil
+}
+
+// Generates a random 16 chars long password that contains uppercase and lowercase characters and numbers or symbols.
+var generatePassword = func() (string, error) {
+	const (
+		uppercaseSet         = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+		lowercaseSet         = "abcdefghijklmnopqrstuvwxyz"
+		numbersAndSymbolsSet = "0123456789*?@"
+		allCharsSet          = uppercaseSet + lowercaseSet + numbersAndSymbolsSet
+	)
+	uppercase, err := getRandomChars(uppercaseSet, 2)
+	if err != nil {
+		return "", err
+	}
+	lowercase, err := getRandomChars(lowercaseSet, 2)
+	if err != nil {
+		return "", err
+	}
+	numbersOrSymbols, err := getRandomChars(numbersAndSymbolsSet, 2)
+	if err != nil {
+		return "", err
+	}
+	allChars, err := getRandomChars(allCharsSet, 10)
+	if err != nil {
+		return "", err
+	}
+	res := []rune(uppercase + lowercase + numbersOrSymbols + allChars)
+	mrand.Shuffle(len(res), func(i, j int) {
+		res[i], res[j] = res[j], res[i]
+	})
+	return string(res), nil
+}
```

Comment:
No need to generate password for users, it's quite difficult to have a proper password generator and we should avoid it

Reply:
There are other open source modules which provides APIs to generate passwords.